src/components/ColorPicker.tsx

```diff
@@ -42,14 +49,90 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThoughtId = useSelector(state => !!state.cursor && head(state.cursor))
+  const currentThoughtValue = useSelector(
+    state => (!!state.cursor && getThoughtById(state, head(state.cursor))?.value) || '',
+  )
+  const simplePath = useSelector(state => simplifyPath(state, state.cursor!))
   size = size || fontSize * 1.2
 
+  /** A function that adds an alpha channel to a hex color. */
+  const addAlphaToHex = (hex: string) => {
+    if (hex.length === 7) return hex + 'ff'
+    return hex
+  }
+
+  const selected = useMemo(() => {
+    // Define the color and background color regex to get the current color of current thought
+    const colorRegex = /color="(#[0-9a-fA-F]{6})"/g
+    const bgColorRegex = /background-color:\s*(rgb\(\d{1,3},\s?\d{1,3},\s?\d{1,3}\))/g
+    const textHexColor = color ? addAlphaToHex(rgbToHex(color)) : undefined
+    const backHexColor = backgroundColor ? addAlphaToHex(rgbToHex(backgroundColor)) : undefined
+    const selectedTextHexColor = cursorStyle?.color ? addAlphaToHex(rgbToHex(cursorStyle?.color)) : undefined
+    const selectedBackHexColor = cursorStyle?.backgroundColor
+      ? addAlphaToHex(rgbToHex(cursorStyle?.backgroundColor))
+      : undefined
+    if (
+      (cursorStyle?.color === undefined && cursorStyle?.backgroundColor === undefined) ||
+      (selectedTextHexColor === '#ccccccff' && selectedBackHexColor === '#333333ff') ||
+      (selectedTextHexColor === addAlphaToHex(rgbToHex(colors.fg)) &&
+        selectedBackHexColor === addAlphaToHex(rgbToHex(colors.bg)) &&
+        !selection.isThought())
+    ) {
+      const colorMatches = currentThoughtValue.match(colorRegex)
+      let matchColor, match
+      // Get the colors and background colors used in current thought's value
+      const fgColors: Set<string> = new Set()
+      if (colorMatches) {
+        colorMatches.forEach(match => fgColors.add(match.slice(7, -1)))
```

Comment:
Why `match.slice(7, -1)`? We have `addAlphaToHex` and `rgbToHex` helpers in order to make color operations more semantic. This operation could also use a helper method to make this code more understandable.